packages/liveblocks-core/src/http-client.ts

```diff
@@ -1,11 +1,1387 @@
+          }
+        }
+
+        throw error;
+      }
+    }
+  }
+
+  // XXX - Batch store for a room will remain in memory even when a room is removed from client.
+  const getAttachmentUrlsBatchStoreByRoom: Map<
+    string,
+    BatchStore<string, string>
+  > = new Map();
+
+  // XXX - This is not a HTTP method, so it feels incorrect to expose this but one of the hooks relies on this store
+  function getOrCreateAttachmentUrlsStore(
+    roomId: string
+  ): BatchStore<string, string> {
+    let store = getAttachmentUrlsBatchStoreByRoom.get(roomId);
+    if (store === undefined) {
+      const batch = new Batch<string, string>(
+        async (batchedAttachmentIds) => {
+          const attachmentIds = batchedAttachmentIds.flat();
+          const { urls } = await httpClient.post<{
+            urls: (string | null)[];
+          }>(
+            url`/v2/c/rooms/${roomId}/attachments/presigned-urls`,
+            () => getAuthValueForRoom(roomId),
+            {
+              attachmentIds,
+            }
+          );
+
+          return urls.map(
+            (url) =>
+              url ??
+              new Error(
+                "There was an error while getting this attachment's URL"
+              )
+          );
+        },
+        {
+          delay: 50,
+        }
+      );
+
+      store = createBatchStore(batch);
+
+      getAttachmentUrlsBatchStoreByRoom.set(roomId, store);
+    }
+    return store;
+  }
+
+  function getAttachmentUrl(options: { roomId: string; attachmentId: string }) {
+    const batch = getOrCreateAttachmentUrlsStore(options.roomId).getBatch();
+    return batch.get(options.attachmentId);
+  }
+
+  /* -------------------------------------------------------------------------------------------------
+   * Notifications (Room level)
+   * -----------------------------------------------------------------------------------------------*/
+  function getNotificationSettings(options: {
+    roomId: string;
+    signal?: AbortSignal;
+  }): Promise<RoomNotificationSettings> {
+    return httpClient.get<RoomNotificationSettings>(
+      url`/v2/c/rooms/${options.roomId}/notification-settings`,
+      () => getAuthValueForRoom(options.roomId),
+      undefined,
+      {
+        signal: options.signal,
+      }
+    );
+  }
+
+  function updateNotificationSettings(options: {
+    roomId: string;
+    settings: Partial<RoomNotificationSettings>;
+  }): Promise<RoomNotificationSettings> {
+    return httpClient.post<RoomNotificationSettings>(
+      url`/v2/c/rooms/${options.roomId}/notification-settings`,
+      () => getAuthValueForRoom(options.roomId),
+      options.settings
+    );
+  }
+
+  // XXX - Batch for a room will remain in memory even when a room is removed from client.
+  // XXX - Ensure this does not cause issues when the room is remounted, etc.
+  const markInboxNotificationsAsReadBatchByRoom: Map<
+    string,
+    Batch<string, string>
+  > = new Map();
```

Comment:
@nvie @marcbouchenoire This batch was previously created as part of the `createRoom` function and was therefore removed from memory when the `Room` object was removed. We had to move this batch from room to the http client for this refactor. We now have a map of room id to the inbox notifications batch but this means, when a room is unmounted, the batch isn't removed automatically. I am not very familiar with the inner workings of a batch myself, so I was curious if you can think of any consequences of doing this.

Reply:
The batching window for these is fairly small (50ms) so I don't think there are big consequences. Even in the scenario where somehow a batching window is opened then a Room is killed before the window is closed, I think it's still fine the make the batched call to the endpoint?